docs/data-structures/preordertraversal.mdx

```diff
@@ -0,0 +1,209 @@
+---
+title: Preorder traversal of Binary Tree
+slug: /data-structures/preordertraversal
+---
+
+## Introduction
+
+Binary Tree is a type of tree which have atmost two children or we can say atmost degree 2. So to traverse the binary tree according to Depth first search. We have a traversal algorithm called Preorder traversal.
+
+## Explanation 
+In this type of traversal(Preorder traversal). First we have to visit the root then its left part then right part respectively. In this first visit root then its left part that means we have to traverse whole left subtree after that whole right subtree. Preorder: (root->left->right). 
+### Example 1
```

Comment:
@shivammahajancse Below paragraphs look more grammatically correct please edit to this

```markdown
## Introduction

A binary tree is a type of tree which have at most two children or we can say almost degree 2. So to traverse the binary tree according to Depth-first search. We have a traversal algorithm called Preorder traversal.

## Explanation

In this type of traversal(Preorder traversal). First, we have to visit the root then the left part then the right part respectively. In this first visit root then its left part that means we have to traverse the whole left subtree after that whole right subtree. Preorder: (root->left->right).
```